BrowserKit/Sources/SiteImageView/URLProcessing/NetworkRequest.swift

```diff
@@ -0,0 +1,34 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import Foundation
+
+protocol NetworkRequest {
+    func fetchDataForURL(_ url: URL, completion: @escaping ((Result<Data, SiteImageError>) -> Void))
+}
+
+class HTMLDataRequest: NetworkRequest {
+
+    enum RequestConstants {
+        static let timeout: TimeInterval = 5
+        static let userAgent = ""
```

Comment:
Empty userAgent for now? What's the plan for this?

Reply:
This is what FaviconFetcher does today, but I just checked and the more commonly used SQLiteHistoryFavicons uses the desktop UA so I will see how I can change it to use that instead.